### PR TITLE
feat(auth): transparent refresh on 401, richer diagnostics, --password-stdin

### PR DESCRIPTION
**Pull request by zjpiazza**
Closes #6.

## Summary

- **Transparent refresh on 401**: `DirectusClient.request()` accepts an `onRefresh` callback. `BaseCommand` wires it to call `refreshAndStoreTokens(profile)` and retry the request once with the new token. Applies to every SDK-backed command.
- **Proactive refresh window**: `isTokenExpired` now considers tokens stale within 60 s of expiry (configurable via `windowMs`). Protects long-running commands and agent workflows from mid-flight expiries.
- **Structured refresh errors**: new `refreshAndStoreTokensDetailed()` returns a `RefreshResult` with a `reason` of `missingProfile | missingRefreshToken | networkError | rejected | unknown`. `auth refresh` surfaces actionable per-reason messages.
- **Richer `auth status`**: emits structured JSON/table output with the auth kind (`static` vs `oauth`), expiry, account (id / email / role), and classified errors (`auth | network | unknown`). Distinguishes "token rejected" from "server unreachable".
- **Non-interactive login**: `auth login --password-stdin` reads the password from stdin for CI and scripted use; mutually exclusive with `--password`.
- **Config hardening**: config file is written with `0600` and its parent dir created with `0700` on POSIX (no-op on Windows). `updateProfileTokens` no longer stores empty-string refresh tokens and preserves the existing token when the key is omitted.

## Tests

- New `test/lib/auth.test.ts` (14 tests): `isTokenExpired` window behaviour + all `refreshAndStoreTokensDetailed` reason paths.
- Extended `test/lib/config.test.ts` (+ 13 tests): chmod 0600 on POSIX, no chmod on Windows, 0700 dir mode, `updateProfileTokens` refresh-token preserve / update / delete semantics, `setDefaultProfile`, `setProfile`, `removeProfile` edge cases.
- Full suite: 64 passing, build + lint clean.

### CODE REVIEW

File: src/lib/auth.ts [hunk 1]
```diff
@@ -153,45 +170,80 @@ export async function logoutAndClearTokens(profileName: string): Promise<void> {
...
 
-      if (!tokens) {
-        return false;
-      }
+    return {ok: true};
+  } catch (error) {
+    return classifyRefreshError(error);
+  } finally {
+    await client.destroy();
+  }
```
[the remark this answers is not in the document]
Comment (zjpiazza) [reply src/lib/auth.ts]: Good catch — fixed in d482361. The Bottleneck limiter in `src/lib/client.ts` is module-scoped and shared across all `DirectusClient` instances, so destroying it from the refresh helper (invoked via `onRefresh` on an in-flight client request) would disconnect the shared limiter and break subsequent `limiter.schedule(...)` calls. Removed `client.destroy()` from both `refreshAndStoreTokensDetailed` and `logoutAndClearTokens`; the limiter has no per-instance state so these short-lived helper clients are safe to leak.

File: src/commands/auth/login.ts [hunk 2]
```diff
@@ -36,8 +42,29 @@ export default class AuthLogin extends Command {
...
+ * Trims trailing whitespace/newlines so `echo "secret" | ...` works as expected.
+ */
+async function readStdin(): Promise<string> {
+  const chunks: Buffer[] = [];
+  for await (const chunk of process.stdin) {
+    chunks.push(Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk as string));
+  }
+
+  return Buffer.concat(chunks).toString('utf8').replace(/\s+$/u, '');
+}
```
[the remark this answers is not in the document]
Comment (zjpiazza) [reply src/commands/auth/login.ts]: Fixed in 127c2dc. `readStdin()` now checks `process.stdin.isTTY` and throws a clear error pointing the user at piping (`echo ... | ...` / `cat secret.txt | ...`) before attempting to drain stdin, so the command no longer hangs in interactive terminals.
